wpinet/src/main/native/cpp/WebServer.cpp

```diff
@@ -289,39 +289,51 @@ void MyHttpConnection::ProcessRequest() {
             wpi::json{{"dirs", std::move(dirs)}, {"files", std::move(files)}}
                 .dump());
       } else {
-        wpi::StringMap<std::string> dirs;
-        wpi::StringMap<std::string> files;
-        for (auto&& entry : fs::directory_iterator{fullpath}) {
-          bool subdir = entry.is_directory(ec);
-          std::string name = entry.path().filename().string();
-          wpi::SmallString<128> nameUriBuf, nameHtmlBuf;
-          if (subdir) {
-            dirs.emplace(
-                name, fmt::format(
-                          "<tr><td><a href=\"{}/\">{}/</a></td><td></td></tr>",
-                          EscapeURI(name, nameUriBuf),
-                          EscapeHTML(name, nameHtmlBuf)));
-          } else {
-            files.emplace(
-                name, fmt::format(
-                          "<tr><td><a href=\"{}\">{}</a></td><td>{}</td></tr>",
-                          EscapeURI(name, nameUriBuf),
-                          EscapeHTML(name, nameHtmlBuf), entry.file_size(ec)));
+        fs::path indexpath = fmt::format("{}index.html", fullpath);
+        if (fs::exists(indexpath)) {
```

Comment:
Instead of nesting, could do it inline so this is an else if and the original else remains rather than doing another indentation level.